refactor(ingest): Auto report workunits

asikowitz opened the pull request:
Will change base once https://github.com/datahub-project/datahub/pull/8051 is merged

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

Commit:
fix(search): enter key with no query should search all (#8036)

xiphl committed:
feat(ingest): Allow csv-enricher to update more types (#7932)

Co-authored-by: xiphl <xiphlerl9@gmail.com>

Commit:
fix(search): only show explore all btn on search and home (#8047)

asikowitz committed:
add deprecations

asikowitz committed:
refactor(ingest): Auto report workunits

hsheth2 committed:
fix(ingest/dbt): fix dbt subtypes for sources (#8048)

mayurinehate committed:
fix(ingest/bigquery): update usage audit log query to include create/drop operations (#7995)

yoonhyejin committed:
feat(docs): add guide on integration ML system via SDKs (#8029)

Co-authored-by: socar-dini <dini@socar.kr>

hsheth2 commented on metadata-ingestion/src/datahub/emitter/mcp.py:
```diff
@@ -220,4 +222,8 @@ def as_workunit(self) -> "MetadataWorkUnit":
                 id=f"{self.entityUrn}-{self.aspectName}-{ts}", mcp=self
             )
```
also need to preserve treat_errors_as_warnings here

asikowitz replied:
Nice catch

asikowitz committed:
refactor(ingest): Make get_workunits() return MetadataWorkUnits (#8051)

- Deprecates UsageAggregationClass, /usageStats?action=batchIngest, UsageStatsWorkUnit
- Removes parsing of UsageAggregationClass in file source, all sinks, and WorkUnitRecordExtractor

hsheth2 commented on metadata-ingestion/src/datahub/ingestion/api/source.py:
```diff
@@ -155,8 +185,30 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "Source":
+            auto_status_aspect,
+            partial(auto_materialize_referenced_tags, active=True),
+            partial(auto_workunit_reporter, self.get_report()),
+            partial(auto_stale_entity_removal, self.stale_entity_removal_handler)
```
this should be done in the stateful ingestion source base, not in the source base class

asikowitz replied:
Did not mean to include these changes, oops. Specifically wanted this PR to be a straight refactor with no true changes, and why I manually call `auto_workunit_reporter` in each source. Let me remove this + the `stale_entity_removal_handler = None` stuff

hsheth2 commented on metadata-ingestion/src/datahub/ingestion/source/aws/glue.py:
```diff
@@ -859,21 +857,18 @@ def get_profile_if_enabled(
-                        )
+                        yield self._create_profile_mcp(
+                            mce, table_stats, column_stats, partition_spec
+                        ).as_workunit()
```
we could use the `auto_workunit` helper to remove the as_workunit calls too

asikowitz replied:
It'd be nice to emit mcps rather than workunits, but I'd rather not put `Union[MetadataWorkUnit, MetadataChangeProposalWrapper]` all over the place. Until we can emit only mcps would prefer to just call `.as_workunit()`. Open to discussion tho

hsheth2 commented on metadata-ingestion/src/datahub/ingestion/source/elastic_search.py:
```diff
@@ -276,6 +277,8 @@ class ElasticsearchSource(Source):
     - Column types associated with each index field
     """
 
+    stale_entity_removal_handler = None
```
don't think we want these littered all over the place

hsheth2 reviewed (commented):
couple initial comments

mayurinehate committed:
refractor(classification): simplify classification handler (#8056)

shirshanka committed:
feat: Add support for Data Products (#8039)

Co-authored-by: Chris Collins <chriscollins3456@gmail.com>

shirshanka committed:
fix(build): fix lint issue (#8066)

david-leifker committed:
feat(system-update): remove datahub-update requirement on schema reg (#7999)

asikowitz committed:
pr feedback

Commit:
fix(gitignore): update gitignore for generated files (#7940)

shubhamjagtap639 committed:
feat(ingestion/kafka): add description in dataset properties (#7974)

Co-authored-by: Harshal Sheth <hsheth2@gmail.com>
Co-authored-by: MohdSiddiqueBagwan <mohdsiddique.bagwan@gslab.com>
Co-authored-by: mohdsiddique <mohdsiddiquebagwan@gmail.com>

siddiquebagwan committed:
fix(ingestion/tableau): ingest parent project name in container properties (#8030)

Co-authored-by: MohdSiddiqueBagwan <mohdsiddique.bagwan@gslab.com>

asikowitz committed:
refactor(ingest): Move source_helpers.py from datahub/utilities -> datahub/api (#8052)

matwalk committed:
fix(ingest/snowflake): lowercase user urn when using email (#7767)

Co-authored-by: mathilde.oustlant <mathilde.oustlant@cdiscount.com>

mayurinehate committed:
fix(ingest/tableau): don't use unsupported sql condition field (#8065)

Co-authored-by: Harshal Sheth <hsheth2@gmail.com>

hsheth2 committed:
fix(ingest/looker): don't prematurely show connectivity success (#8070)

rinzool committed:
feat(web): update AWS logos (#8057)

david-leifker committed:
fix(metadata-io): remove assert in favor of exceptions (#8035)

yoonhyejin committed:
feat: add docs on  column-level linage (#8062)

hsheth2 committed:
ci: prevent qodana from using all of our cache (#8054)

hsheth2 committed:
ci(ingest/clickhouse): don't use kernel ephemeral ports (#8060)

hsheth2 committed:
test(sdk): better error messages in registry codegen test (#8081)

anshbansal committed:
doc(managed datahub): update release notes for 0.2.7 (#8088)

treff7es committed:
feat(ingest/s3) - Stateful ingestion and last-updated support (#8022)

hsheth2 committed:
docs(ingest/snowflake): fix authentication type docs (#8059)

alplatonov committed:
fix(ingest/s3_data_lake)_ingestor_skips_directories_with_similar_prefix (#8078)

Co-authored-by: Tamas Nemeth <treff7es@gmail.com>

chriscollins3456 committed:
fix(ui) Fix entity name styling to show deprecation and others properly (#8084)

hsheth2 committed:
test(sdk): move cli tests into the unit dir (#8028)

hsheth2 committed:
feat(sdk): better auth error messages in the rest emitter (#8025)

gabe-lyons committed:
feat(caching): skip cache on ownership tabs (#8082)

Commit:
feat(embed): embed lookup route (#8033)

asikowitz committed:
fix(ingest/delta-lake): Walk through directory structure with full path; reduce resource creation (#8072)

iprentic committed:
feat(search): Add AggregateAcrossEntities endpoint (#8000)

RyanHolstien committed:
chore(vulnerability): add exclusions for json to prevent leaking dependency (#8090)

asikowitz committed:
Merge branch 'master' into auto-report-workunits
